components/raftstore/src/store/peer_storage.rs

```diff
@@ -2118,7 +2118,7 @@ mod tests {
     ) -> PeerStorage<KvTestEngine, RaftTestEngine> {
         let mut store = new_storage(region_scheduler, raftlog_fetch_scheduler, path);
         let mut write_task = WriteTask::new(store.get_region_id(), store.peer_id, 1);
-        store.append(ents[1..].to_vec(), &mut write_task);
+        store.append(ents[0..].to_vec(), &mut write_task);
```

Comment:
Why? The first entry is just a placeholder.

Reply:
I don't understand

Reply:
The first entry don't need to be stored. Only its index and term are used to initialized truncated state. The test logic is from tikv/raft-rs.

Reply:
Seems weird to me.. But I'll restore it.

Reply:
Actually, it won't work with raft engine. Raft Engine's `first_index()` is directly deduced from the stored entries.

Reply:
So there will be at least one entry left even if all of them are deleted by raft? What if it catches logs by snapshot? There will be no stored entries.

Reply:
Turns out `first_index` is not a trait interface. I'll try restoring them again.

Reply:
It doesn't work. In some test cases we try to append [3,5] entries to a region of `new_storage_from_ents([3,5])`, but Raft Engine will panic if we append [3,5] to [4,5].